### PR TITLE
add set(difference) function

### PR DESCRIPTION
**Pull request by Abdulhaffiz-Umar**
#1558

### CODE REVIEW

**Comment by Daniel4078** [comment]
I suppose that you should first create an issue for the sub task that you choose before you produce a pull request, in that way, we can track the progress of our TODO list more easily. Thank you.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tf_functions.py [hunk 1]
```diff
@@ -224,3 +224,36 @@ def test_tensorflow_subtract(
...
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_name="difference",
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tf_functions.py:256]: Please refer to this pull request (https://github.com/unifyai/ivy/pull/2903) as fn_name should be changed to fn_tree. Also please add a blank line at the end of the file to conform to formatting standards. Thank you.

File: ivy/functional/frontends/tensorflow/functions.py [hunk 2]
```diff
@@ -32,3 +32,10 @@ def subtract(x, y, name=None):
 
 
 subtract.unsupported_dtypes = {"torch": ("float16", "bfloat16")}
+
+
+def difference(x, y, aminusb=True, validate_indices=True): 
+    return ivy.difference(x,y)
+
+
+subtract.unsupported_dtypes = {"torch": ("float16", "bfloat16")}
```
Comment (Daniel4078) [inline ivy/functional/frontends/tensorflow/functions.py:41]: I don't think ivy.difference is defined in the code base, so it is not the correct function to use here.
Please add a blank line at the end of the file to conform to formatting standards.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
There are some changes to be made before the pull request can be approved, Thank you.

**Comment by Daniel4078** [comment]
Also, ivy.difference does not exist yet and your current implementation is wrong. Please read the last paragraph here:
https://lets-unify.ai/ivy/contributing/4_open_tasks.html#frontend-apis
You might need to convert this issue.

**Comment by Abdulhaffiz-Umar** [comment]
> Also, ivy.difference does not exist yet and your current implementation is wrong. Please read the last paragraph here: https://lets-unify.ai/ivy/contributing/4_open_tasks.html#frontend-apis You might need to convert this issue.

Thank you for the feedback, which branch should I channel the conversion to?

**Comment by Daniel4078** [comment]
In the case where your chosen function exists in all frameworks by default, but is not implemented in Ivy’s functional API, please convert your existing GitHub issue to request for the function to be added to Ivy. Meanwhile, you can select another frontend function to work on from the ToDo list!

**Comment by Daniel4078** [comment]
After you change the issue to be requesting adding the ivy.difference (set-version) function, you can close this pull request and start working on another TODO List issue.

**Pull request closed without merging**